src/GlobalScreen/Scope/MainMenu/Collector/Renderer/BaseTypeRenderer.php

```diff
@@ -30,13 +30,20 @@ class BaseTypeRenderer implements TypeRenderer
      */
     protected $ui_factory;
 
+    /**
+     * @var Renderer
+     */
+    protected $ui_renderer;
+
+
     /**
      * BaseTypeRenderer constructor.
      */
     public function __construct()
     {
         global $DIC;
         $this->ui_factory = $DIC->ui()->factory();
+        $this->ui_renderer = $DIC->ui()->renderer();
```

Comment:
where is this needed?

Reply:
I currently need a legacy slate to inject a DD in the menu: https://github.com/ILIAS-eLearning/ILIAS/pull/3726/files#diff-7deee1ec3fe029b2adca88f0f292dea0f8599c3715a6f394cc4da916b3dd1665R27

Hoever, with a (yet to build) DD-Slate, this surely could and should go away again.